lib/Lower/Bridge.cpp

```diff
@@ -627,119 +720,214 @@ class FirConverter : public Fortran::lower::AbstractConverter {
+                  Fortran::parser::ForallConstruct> &b) {
+                genFIR(eval, b.value());
+              },
               [&](const Fortran::parser::Statement<Fortran::parser::ForallStmt>
                       &b) {
                 setCurrentPosition(b.source);
-                genFIR(b.statement);
+                genFIR(eval, b.statement);
               },
           },
           s.u);
     }
     TODO();
   }
-  void genFIR(const Fortran::parser::ForallAssignmentStmt &s) {
-    std::visit([&](auto &b) { genFIR(b); }, s.u);
-  }
-
-  void genFIR(const Fortran::parser::CompilerDirective &) { TODO(); }
-  void genFIR(const Fortran::parser::OpenMPConstruct &) { TODO(); }
-  void genFIR(const Fortran::parser::OmpEndLoopDirective &) { TODO(); }
-
-  void genFIR(const Fortran::parser::AssociateStmt &) { TODO(); }
-  void genFIR(const Fortran::parser::EndAssociateStmt &) { TODO(); }
-  void genFIR(const Fortran::parser::BlockStmt &) { TODO(); }
-  void genFIR(const Fortran::parser::EndBlockStmt &) { TODO(); }
-  void genFIR(const Fortran::parser::SelectCaseStmt &) { TODO(); }
-  void genFIR(const Fortran::parser::CaseStmt &) { TODO(); }
-  void genFIR(const Fortran::parser::EndSelectStmt &) { TODO(); }
-  void genFIR(const Fortran::parser::ChangeTeamStmt &) { TODO(); }
-  void genFIR(const Fortran::parser::EndChangeTeamStmt &) { TODO(); }
-  void genFIR(const Fortran::parser::CriticalStmt &) { TODO(); }
-  void genFIR(const Fortran::parser::EndCriticalStmt &) { TODO(); }
-
-  // Do loop is handled by EvalIterative(), EvalStructuredOp()
-  void genFIR(const Fortran::parser::NonLabelDoStmt &) {} // do nothing
-  void genFIR(const Fortran::parser::EndDoStmt &) {}      // do nothing
-
-  // If-Then-Else is handled by EvalCondGoto(), EvalStructuredOp()
-  void genFIR(const Fortran::parser::IfThenStmt &) {} // do nothing
-  void genFIR(const Fortran::parser::ElseIfStmt &) {} // do nothing
-  void genFIR(const Fortran::parser::ElseStmt &) {}   // do nothing
-  void genFIR(const Fortran::parser::EndIfStmt &) {}  // do nothing
-
-  void genFIR(const Fortran::parser::SelectRankStmt &) { TODO(); }
-  void genFIR(const Fortran::parser::SelectRankCaseStmt &) { TODO(); }
-  void genFIR(const Fortran::parser::SelectTypeStmt &) { TODO(); }
-  void genFIR(const Fortran::parser::TypeGuardStmt &) { TODO(); }
-
-  void genFIR(const Fortran::parser::WhereConstructStmt &) { TODO(); }
-  void genFIR(const Fortran::parser::MaskedElsewhereStmt &) { TODO(); }
-  void genFIR(const Fortran::parser::ElsewhereStmt &) { TODO(); }
-  void genFIR(const Fortran::parser::EndWhereStmt &) { TODO(); }
-  void genFIR(const Fortran::parser::ForallConstructStmt &) { TODO(); }
-  void genFIR(const Fortran::parser::EndForallStmt &) { TODO(); }
+  void genFIR(Fortran::lower::pft::Evaluation &eval,
+              const Fortran::parser::ForallAssignmentStmt &s) {
+    // std::visit([&](auto &b) { genFIR(b); }, s.u);  // ...
```

Comment:
Bug?

Reply:
Yes, thanks.  This one slipped through the cracks.